### PR TITLE
Add a way to tell if RemoteControlClient has an active connection

### PR DESCRIPTION
**Pull request by mrlacey**
GitHub Issue (If applicable): 

## PR Type

What kind of change does this PR introduce?

- Other: small API change/addition (in a very niche tooling area) to aid debuggability and reliability of complex inter-proc communication scenarios.

## What is the current behavior?

There's no way for anything using RemoteControlClient to know if it has successfully connected. 
This means that anything other than HotReload that wishes to use the RemoteControl system cannot know if a successful connection to the server has been made.

## What is the new behavior?

Add a public property that consumers can use to tell if a connection to the server exists.
This allows them to know if it's worth trying to send a message.

### <samp>🤖 Generated by Copilot at fa4b65e</samp>

Added `IsConnected` property to `RemoteControlClient` to expose connection status. This is part of a pull request to improve the remote control feature of Uno Platform.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
-  Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
-  Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Connected to #13088 

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RemoteControl/RemoteControlClient.cs [hunk 1]
```diff
@@ -76,6 +76,8 @@ private RemoteControlClient(Type appType)
 			_ = StartConnection();
 		}
 
+		public bool IsConnected => _webSocket is not null;
```
Comment (ajpinedam) [inline src/Uno.UI.RemoteControl/RemoteControlClient.cs:79]: I see once the connection is made, we never close it, so this should never be `null` once the connection is established.

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI.RemoteControl/RemoteControlClient.cs (hunk 1 above)
Comment (mrlacey) [reply src/Uno.UI.RemoteControl/RemoteControlClient.cs:79]: At the moment, I'm using this as a way to detect if a connection has been established, and so closing isn't a concern.
The connection is closed when a debug session finishes.
If something forces the connection to break, the current behavior/requirement is to restart the debug session to use the functionality requiring RemoteControl.
If (and when) needed, we can add the capability to restart the connection without restarting the debug session later.

Comment (jeromelaban) [reply src/Uno.UI.RemoteControl/RemoteControlClient.cs:79]: There's a similar change in https://github.com/unoplatform/uno/pull/12846, (https://github.com/unoplatform/uno/pull/12846/files#diff-0a4e1397a5062960f20dea694ab5858fafa2ebd5973737aca619d8869f62814aR90), with `WaitForConnection`. Do you need to asynchronously wait for client to be connected, or is it for something else?

Comment (mrlacey) [reply src/Uno.UI.RemoteControl/RemoteControlClient.cs:79]: Looks like we're both trying to address the same issue. 
I wanted a quick way to check when the RCClient had a connection before I started sending lots of messages through it.
I was looking for a way to make minimal changes to RCClient but if there are separate bigger changes happening there I can wait for them.

**Comment by mrlacey** [comment]
Reverting to draft so I can more easily track this and find a temporary solution until #12846 is merged.
I'll close this once I have a temporary workaround

**Pull request closed without merging**